src/caps.rs

```diff
@@ -1,43 +1,248 @@
-use std::time::Duration;
+use std::{collections::BTreeSet, str::FromStr, time::Duration};
 
-use anyhow::{Context, Result};
-use ed25519_dalek::{SigningKey, VerifyingKey};
+use anyhow::{bail, Context, Result};
+use ed25519_dalek::SigningKey;
 use iroh::NodeId;
 use rcan::{Capability, Expires, Rcan};
 use serde::{Deserialize, Serialize};
 use ssh_key::PrivateKey as SshPrivateKey;
 
-#[derive(Ord, Eq, PartialOrd, PartialEq, Clone, Serialize, Deserialize, Debug)]
-pub enum IpsCap {
-    V1(IpsCapV1),
+macro_rules! cap_enum(
+    ($enum:item) => {
+        #[derive(
+            Debug,
+            Eq,
+            PartialEq,
+            Ord,
+            PartialOrd,
+            Serialize,
+            Deserialize,
+            Clone,
+            Copy,
+            strum::Display,
+            strum::EnumString,
+        )]
+        #[strum(serialize_all = "kebab-case")]
+        #[serde(rename_all = "kebab-case")]
+        $enum
+    }
+);
+
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize, Clone)]
+pub enum Caps {
+    V0(CapSet<Cap>),
+}
+
+impl Default for Caps {
+    fn default() -> Self {
+        Self::V0(CapSet::default())
+    }
+}
+
+impl std::ops::Deref for Caps {
+    type Target = CapSet<Cap>;
+
+    fn deref(&self) -> &Self::Target {
+        let Self::V0(ref slf) = self;
+        slf
+    }
+}
+
+#[derive(
+    Debug,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Serialize,
+    Deserialize,
+    Clone,
+    Copy,
+    derive_more::From,
+    strum::Display,
+)]
+pub enum Cap {
+    #[strum(to_string = "all")]
+    All,
+    #[strum(to_string = "blobs:{0}")]
+    Blobs(BlobsCap),
+    #[strum(to_string = "relay:{0}")]
+    Relay(RelayCap),
+    #[strum(to_string = "metrics:{0}")]
+    Metrics(MetricsCap),
+}
+
+impl FromStr for Cap {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        if s == "all" {
+            Ok(Self::All)
+        } else if let Some((domain, inner)) = s.split_once(":") {
+            Ok(match domain {
+                "blobs" => Self::Blobs(BlobsCap::from_str(inner)?),
+                "metrics" => Self::Metrics(MetricsCap::from_str(inner)?),
+                "relay" => Self::Relay(RelayCap::from_str(inner)?),
+                _ => bail!("invalid cap domain"),
+            })
+        } else {
+            Err(anyhow::anyhow!("invalid cap string"))
+        }
+    }
+}
+cap_enum!(
+    pub enum BlobsCap {
+        All,
+        PutBlob,
+        GetTag,
```

Comment:
lets also add  `GetBlob`, as limiting that is simply a todo atm, but really needs to be done